validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/OkHttpValidatorRestApiClient.java

```diff
@@ -0,0 +1,259 @@
+      final UInt64 slot, final int committeeIndex) {
+    final Map<String, String> queryParams = new HashMap<>();
+    queryParams.put("slot", encodeQueryParam(slot));
+    queryParams.put("committee_index", String.valueOf(committeeIndex));
+
+    return get(GET_UNSIGNED_ATTESTATION, queryParams, Attestation.class);
+  }
+
+  @Override
+  public void sendSignedAttestation(final Attestation attestation) {
+    post(SEND_SIGNED_ATTESTATION, attestation, null);
+  }
+
+  @Override
+  public Optional<Attestation> createAggregate(final Bytes32 attestationHashTreeRoot) {
+    final Map<String, String> queryParams = new HashMap<>();
+    queryParams.put("slot", encodeQueryParam(UInt64.ZERO));
+    queryParams.put("attestation_data_root", encodeQueryParam(attestationHashTreeRoot));
+
+    return get(GET_AGGREGATE, queryParams, Attestation.class);
+  }
+
+  @Override
+  public void sendAggregateAndProof(final SignedAggregateAndProof signedAggregateAndProof) {
+    post(SEND_SIGNED_AGGREGATE_AND_PROOF, signedAggregateAndProof, null);
+  }
+
+  @Override
+  public void subscribeToBeaconCommitteeForAggregation(
+      final int committeeIndex, final UInt64 aggregationSlot) {
+    final SubscribeToBeaconCommitteeRequest request =
+        new SubscribeToBeaconCommitteeRequest(committeeIndex, aggregationSlot);
+    post(SUBSCRIBE_TO_COMMITTEE_FOR_AGGREGATION, request, null);
+  }
+
+  @Override
+  public void subscribeToPersistentSubnets(final Set<SubnetSubscription> subnetSubscriptions) {
+    post(SUBSCRIBE_TO_PERSISTENT_SUBNETS, subnetSubscriptions, null);
+  }
+
+  public <T> Optional<T> get(
+      final ValidatorApiMethod apiMethod,
+      final Map<String, String> queryParams,
+      final Class<T> responseClass) {
+    final HttpUrl.Builder httpUrlBuilder = urlBuilder(apiMethod);
+    if (queryParams != null && !queryParams.isEmpty()) {
+      queryParams.forEach(httpUrlBuilder::addQueryParameter);
+    }
+
+    final Request request = new Request.Builder().url(httpUrlBuilder.build()).build();
+    return executeCall(request, responseClass);
+  }
+
+  private <T> Optional<T> post(
+      final ValidatorApiMethod apiMethod,
+      final Object requestBodyObj,
+      final Class<T> responseClass) {
+    final String requestBody;
+    try {
+      requestBody = jsonProvider.objectToJSON(requestBodyObj);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    final Request request =
+        new Request.Builder()
+            .url(urlBuilder(apiMethod).build())
+            .post(RequestBody.create(requestBody, APPLICATION_JSON))
+            .build();
+
+    return executeCall(request, responseClass);
+  }
+
+  private HttpUrl.Builder urlBuilder(final ValidatorApiMethod apiMethod) {
+    return new HttpUrl.Builder()
+        .scheme(baseEndpoint.scheme())
+        .host(baseEndpoint.host())
+        .port(baseEndpoint.port())
+        .encodedPath(apiMethod.getPath());
```

Comment:
Maybe:
```suggestion
  private HttpUrl.Builder urlBuilder(final ValidatorApiMethod apiMethod) {
    return baseEndpoint.resolve(apiMethod.getPath());
```

That way if the base URL includes a username/password or other elements we didn't expect they'll be preserved.  Slightly tempted to make the paths in `ValidatorApiMethod` not start with `/` so they're fully relative and then if you specified `http://localhost:5051/beaconNode/` we'd preserve the `/beaconNode/` path which may be useful when load balancers get involved or to use a future Infura API where a product key would be included in the URL.